iris/tests/download_manager/change_download_folder.py

```diff
@@ -0,0 +1,160 @@
+
+        # Navigate to about:preferences.
+        navigate('about:preferences#searchResults')
+
+        expected = exists(AboutPreferences.ABOUT_PREFERENCE_SEARCH_PAGE_PATTERN, 10)
+        assert_true(self, expected, 'The \'about:preferences#search\' page successfully loaded.')
+
+        time.sleep(DEFAULT_UI_DELAY_LONG)
+
+        type('downloads')
+        expected = exists(AboutPreferences.DOWNLOADS, 10)
+        assert_true(self, expected, 'The \'Downloads\' section is displayed.')
+
+        click(AboutPreferences.BROWSE)
+        expected = exists(AboutPreferences.DOWNLOADS, 10)
+        assert_true(self, expected, '\'New Folder\' button is displayed.')
+
+        click(Utils.NEW_FOLDER)
+        expected = exists(Utils.NEW_FOLDER_HIGHLIGHTED, 10)
+        assert_true(self, expected, '\'New Folder\' is highlighted.')
+
+        type('new_downloads_folder')
+
+        type(Key.ENTER)
+        expected = exists(Utils.NEW_DOWNLOADS_FOLDER_HIGHLIGHTED, 10)
+        assert_true(self, expected, '\'New Downloads Folder\' is created.')
+
+        click(Utils.SELECT_FOLDER)
+
+        try:
+            expected_3 = wait_vanish(Utils.SELECT_FOLDER, 10)
+            assert_true(self, expected_3, 'Downloads folder option window is dismissed.')
+        except FindError:
+            logger.error('Downloads folder option is still displayed.')
+
+        navigate(LocalWeb.THINKBROADBAND_TEST_SITE)
+
+        # Wait for the page to be loaded.
+        try:
+            wait(DownloadFiles.SMALL_FILE_10MB, 10)
+            logger.debug('File is present in the page.')
+        except FindError:
+            raise FindError('File is not present in the page.')
+
+        download_file(DownloadFiles.SMALL_FILE_10MB, DownloadFiles.OK)
+
+        expected = exists(NavBar.DOWNLOADS_BUTTON_BLUE, 10)
+        assert_true(self, expected, 'Downloads button turns blue.')
+
+        click(NavBar.DOWNLOADS_BUTTON_BLUE)
+
+        expected = exists(DownloadFiles.DOWNLOADS_PANEL_5MB_COMPLETED, 10)
+        assert_true(self, expected, 'The 5MB download is complete.')
+
+        file_5_mb = find(DownloadFiles.DOWNLOADS_PANEL_5MB_COMPLETED)
+        region_5_mb = Region(file_5_mb.x, file_5_mb.y - 10, 500, 50)
+        expected = region_5_mb.exists(DownloadManager.DownloadsPanel.OPEN_CONTAINING_FOLDER, 10)
+        assert_true(self, expected, 'Containing folder icon is available for 5mb file.')
+
+        click(DownloadManager.DownloadsPanel.OPEN_CONTAINING_FOLDER, in_region=region_5_mb)
+        time.sleep(DEFAULT_UI_DELAY_LONG)
+
+        expected = exists(DownloadManager.DOWNLOADS_FOLDER, 10)
+        assert_true(self, expected, 'Default Downloads folder is displayed.')
+
+        expected = exists(DownloadFiles.FOLDER_VIEW_5MB_HIGHLIGHTED, 10)
+        assert_true(self, expected, 'Downloaded file is found.')
+
+        close_tab()
+
+        # Refocus the firefox window.
+        click(NavBar.HOME_BUTTON.target_offset(70, 0))
```

Comment:
yes, can work, but won't work all the time. 
Color accuracy sometimes differs for the windows that are not in focus, so in order to avoid this pattern matching issues, I would rather prefer to make sure that I have the page in focus before searching for a specific pattern, in that way the test will be more stable as well while working with image patterns searching.